src/Nethermind/Nethermind.Consensus/Withdrawals/BlockProductionWithdrawalProcessor.cs

```diff
@@ -22,7 +22,7 @@ public void ProcessWithdrawals(Block block, IReleaseSpec spec)
 
         if (spec.WithdrawalsEnabled)
         {
-            block.Header.WithdrawalsRoot = block.Withdrawals is null || block.Withdrawals.Length == 0
+            block.Header.WithdrawalsRoot = block.Withdrawals!.Length == 0
```

Comment:
Looks risky, it can be null right?

Reply:
This is not for review now

Reply:
Can this code be reached without this validation from tests or something?

Reply:
Nope. That's the intention at least. When withdrawals are enabled for a given block, `Block.Withdrawals` must not be `null`. If it's null, then we're missing validation somewhere. So, with this change, I'm trying to catch that instead of silently processing the invalid block which may lead to a hardly identifiable unexpected behavior.

Reply:
I would suggest not changing it. It is better to handle this null. If we miss this validation (and we had such bugs before) the consequences will be more lightweight when we have a proper null handling. I don't see the value of changing it.